eng/referenceFromRuntime.targets

```diff
@@ -31,20 +31,35 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="GetFilesFromCoreCLR" Returns="@(CoreCLRFiles)" DependsOnTargets="ResolveCoreCLRFilesFromLocalBuild" />
+  <Target Name="GetFilesFromRuntime" Returns="@(RuntimeFiles)" DependsOnTargets="GetFilesFromCoreClr;GetFilesFromMono">
+    <ItemGroup>
+      <RuntimeFiles Include="@(CoreCLRFiles)" Condition="'$(RuntimeFlavor)' == 'CoreCLR'" />
+      <RuntimeFiles Include="@(MonoFiles)" Condition="'$(RuntimeFlavor)' == 'Mono'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GetFilesFromCoreCLR" 
+          Condition="'$(RuntimeFlavor)' == 'CoreCLR'"
```

Comment:
I don't see RuntimeFlavor being set to CoreCLR anywhere. Should this be != Mono instead?

Reply:
That's interesting and feels like something we should be fixing. In general using the absence of one property to imply a scenario is enabled is a bit of "MSBuild code smell" for me. Much better to use explicit values to declare a scenario is supported. Makes it easier to detect when build logic isn't properly flowing values. 

In the short term though will take the suggestion cause I don't want to explode this PR. Want to focus on getting the Mono scenario working.

Reply:
yeah I think it makes sense to be explicit but the current change would break CoreCLR since nothing sets RuntimeFlavor=CoreCLR :)

Reply:
Isn't the canonical way to solve this by slapping the following code somewhere at the top of the file (either referenceFromRuntime.targets or src/libraries/restore/Directory.Build.targets)?

```xml
    <PropertyGroup>
        <!-- Default to CoreCLR runtime flavor if not specified -->
        <RuntimeFlavor Condition="'$(RuntimeFlavor)' == ''">CoreCLR</RuntimeFlavor>
    </PropertyGroup>
```

Reply:
@filipnavara in general yes but the runtime build is involved and can have odd dependencies. Example: last time I did exactly that type of change I spent almost two weeks tracking down the build errors. 😦 

Want to tackle that separately so it doesn't hold up unblocking this Mono scenario. Hopefully it turns out to be that easy.

Reply:
@akoeplinger 

The `$(RuntimeFlavor)` is set to `CoreCLR` in our yml files but apparently not in our standard build. That inconsistency needs to be addressed still. At the same time `!= 'Mono'` is indeed the safest form of the check at the moment.

Reply:
Tracking issue https://github.com/dotnet/runtime/issues/32884